core/src/identity/secp256k1.rs

```diff
@@ -51,6 +52,12 @@ impl Keypair {
     }
 }
 
+impl fmt::Debug for Keypair {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Keypair {{ secret: \"Secret Key\", public: \"{:?}\" }}", &self.public)
```

Comment:
The `Formatter` has helpers for outputting structs:

```suggestion
        f.debug_struct("Keypair").field("public", &self.public).finish()
```